upf/upf_api.c

```diff
@@ -1061,19 +1061,84 @@ vl_api_upf_tdf_ul_table_add_t_handler (vl_api_upf_tdf_ul_table_add_t * mp)
   REPLY_MACRO (VL_API_UPF_TDF_UL_TABLE_ADD_REPLY);
 }
 
+/* API message handler */
+static void
+vl_api_upf_ueip_pool_nwi_add_t_handler (vl_api_upf_ueip_pool_nwi_add_t * mp)
+{
+  int rv = 0;
+  upf_main_t *sm = &upf_main;
+  vl_api_upf_ueip_pool_nwi_add_reply_t *rmp = NULL;
+
+  u8 *nwi_name = mp->nwi_name;
+  u8 *nwi_name_vec = 0;
+  u8 *identity = mp->identity;
+  u8 *identity_vec = 0;
+  u8 identity_len = strlen (identity);
+
+  vec_validate (nwi_name_vec, mp->nwi_name_len);
+  memcpy (nwi_name_vec, nwi_name, mp->nwi_name_len);
+
+  vec_validate (identity_vec, identity_len);
+  memcpy (identity_vec, identity, identity_len);
+
+  rv = vnet_upf_ue_ip_pool_add_del (identity_vec, nwi_name_vec, mp->is_add);
+
+  REPLY_MACRO (VL_API_UPF_UEIP_POOL_NWI_ADD_REPLY);
+}
+
+/* API message handler */
+static void
+vl_api_upf_ueip_pool_dump_t_handler (vl_api_upf_ueip_pool_dump_t * mp)
+{
+  upf_main_t *sm = &upf_main;
+  vl_api_registration_t *reg;
+  vl_api_upf_ueip_pool_details_t *rmp = NULL;
+  uword *p;
+  upf_ue_ip_pool_info_t *pool = NULL;
+  u8 *identity;
+
+  reg = vl_api_client_index_to_registration (mp->client_index);
+  if (!reg)
+    return;
+
+  /* *INDENT-OFF* */
+  hash_foreach (identity, p, sm->ue_ip_pool_index_by_identity,
+    {
+      u8 nwi_name_len;
+      u8 identity_len;
+
+      p = hash_get_mem (sm->ue_ip_pool_index_by_identity, identity);
+      pool = pool_elt_at_index (sm->ueip_pools, p[0]);
+
+      nwi_name_len = strlen (pool->nwi_name);
```

Comment:
Please check that `nwi_name_len` doesn't exceed the max possible length (64).
Best of all, it should be ensured that NWIs which such long names can't be created (incl. via CLI), and an assertion added here